### PR TITLE
Fix jax.scipy.stats.vonmises pdf/logpdf returning nan at kappa=0

### LINKED ISSUE #38621: jax.scipy.stats.vonmises.pdf returns nan at kappa=0.0
### Description

`jax.scipy.stats.vonmises.pdf(x, kappa=0.0)` returns `nan` for finite values of
`x`, while SciPy returns approximately `0.15915494`.

Examples:

```text
vonmises.pdf(0.0,  kappa=0.0): scipy≈0.15915494, jax=nan
vonmises.pdf(1.0,  kappa=0.0): scipy≈0.15915494, jax=nan
vonmises.pdf(-1.0, kappa=0.0): scipy≈0.15915494, jax=nan
```

This appears to be a boundary handling mismatch for the valid parameter value
`kappa=0.0`. In this case, the von Mises distribution reduces to the uniform
distribution on the circle, whose density is `1 / (2π)`.

Looking at the JAX implementation, the `pdf` is computed as:

```text
jnp.where(kappa > 0, <closed-form density>, jnp.nan)
```

So `nan` is returned by an explicit branch whenever `kappa` is not strictly
positive, rather than by an indeterminate expression silently propagating.
`kappa=0` is a mathematically valid, finite parameter value (it is excluded
only by the strict `>` comparison), so this branch appears to treat a valid
boundary case as invalid input.

## Origin

Found via property-based testing derived from:

```text
jax/tests/lax_scipy_stats_test.py
  ::LaxBackedScipyStatsTests::testVonMisesPdf
```

The property compares `jax.scipy.stats.vonmises.pdf` against
`scipy.stats.vonmises.pdf` for special values and valid distribution parameters.

The existing JAX test only forces `kappa` to be non-negative (by taking its
absolute value); `kappa=0` is reachable in principle but essentially never
sampled by the underlying random generator in practice, and `nan`/`inf`
inputs for `x` are never exercised either. Testing explicitly with
`kappa ∈ {0.0, 1.0}` and `x ∈ {nan, inf, -inf, 0.0, 1.0, -1.0}` reproduces the
failure deterministically for every finite `x` when `kappa=0.0`, while
`kappa=1.0` (used as a control) and the `nan`/`inf` cases for `x` behave as
expected in both libraries.

The failing case uses:

```python
x = 0.0
kappa = 0.0
```

The same behavior also appears for other finite values of `x`.

## Verification

The von Mises density is:

```text
f(x; kappa) = exp(kappa * cos(x)) / (2π I0(kappa))
```

where `I0` is the modified Bessel function of order 0.

For `kappa=0.0`:

```text
exp(0 * cos(x)) = 1
I0(0) = 1
```

so the density becomes:

```text
f(x; 0) = 1 / (2π) ≈ 0.15915494309189535
```

This suggests that finite `x` values with `kappa=0.0` should produce a finite
constant density rather than `nan`. SciPy returns values consistent with this
formula.

A direct check with `mpmath` gives the same value:

```python
import mpmath as mp

kappa = mp.mpf("0.0")
x = mp.mpf("0.0")

density = mp.exp(kappa * mp.cos(x)) / (2 * mp.pi * mp.besseli(0, kappa))
print(density)
```

Output:

```text
0.15915494309189533577
```

## Minimal reproducer

```python
import numpy as np
import scipy.stats
import jax
import jax.numpy as jnp
import jax.scipy.stats

jax.config.update("jax_enable_x64", True)

cases = [
    (np.float32(0.0), np.float32(0.0)),
    (np.float64(0.0), np.float64(0.0)),
    (np.float32(1.0), np.float32(0.0)),
    (np.float32(-1.0), np.float32(0.0)),
    (np.float32(np.pi), np.float32(0.0)),
    (np.float32(-np.pi), np.float32(0.0)),

    # Control cases
    (np.float32(0.0), np.float32(1.0)),
    (np.float32(np.nan), np.float32(0.0)),
    (np.float32(np.inf), np.float32(0.0)),
]

for x, kappa in cases:
    scipy_result = scipy.stats.vonmises.pdf(np.array(x), np.array(kappa))
    jax_result = jax.scipy.stats.vonmises.pdf(jnp.array(x), jnp.array(kappa))
    jit_result = jax.jit(
        lambda x_, kappa_: jax.scipy.stats.vonmises.pdf(x_, kappa_)
    )(jnp.array(x), jnp.array(kappa))

    print(f"x={x!r}, kappa={kappa!r}")
    print(f"  scipy: {scipy_result}")
    print(f"  jax:   {jax_result}")
    print(f"  jit:   {jit_result}")
    print()
```

## Expected behavior

For finite `x` and `kappa=0.0`, JAX would be expected to return the constant
density of the uniform distribution on the circle:

```text
x=np.float32(0.0), kappa=np.float32(0.0)
  scipy: 0.15915493866300567
  jax:   0.15915494
  jit:   0.15915494
```

## Actual behavior

```text
x=np.float32(0.0), kappa=np.float32(0.0)
  scipy: 0.15915493866300567
  jax:   nan
  jit:   nan

x=np.float64(0.0), kappa=np.float64(0.0)
  scipy: 0.15915494309189535
  jax:   nan
  jit:   nan

x=np.float32(1.0), kappa=np.float32(0.0)
  scipy: 0.15915493866300567
  jax:   nan
  jit:   nan

x=np.float32(-1.0), kappa=np.float32(0.0)
  scipy: 0.15915493866300567
  jax:   nan
  jit:   nan
```

## Additional cases

```text
x=0.0,  kappa=0.0: scipy≈0.15915494, jax=nan  ← mismatch
x=1.0,  kappa=0.0: scipy≈0.15915494, jax=nan  ← mismatch
x=-1.0, kappa=0.0: scipy≈0.15915494, jax=nan  ← mismatch
x=pi,   kappa=0.0: scipy≈0.15915494, jax=nan  ← mismatch
x=-pi,  kappa=0.0: scipy≈0.15915494, jax=nan  ← mismatch

x=0.0, kappa=1.0: finite values are close within tolerance  OK
x=nan, kappa=0.0: scipy=nan, jax=nan  OK
x=inf, kappa=0.0: scipy=nan, jax=nan  OK
```

The mismatch appears specific to finite `x` with `kappa=0.0`.

### System info (python version, jaxlib version, accelerator, etc.)

```text
jax:     0.10.1
jaxlib:  0.10.1
numpy:   2.4.6
scipy:   1.17.1
python:  3.12.3
OS:      Linux-6.17.0-29-generic-x86_64-with-glibc2.39
backend: cpu
```

### PR DESCRIPTION
**Pull request by RoyK0108**
Fixes #38621.

### Problem

`jax.scipy.stats.vonmises.pdf(x, kappa=0.0)` returns `nan` for finite `x`,
while SciPy returns `1 / (2 * pi) ≈ 0.15915494`. At `kappa == 0` the von Mises
distribution reduces to the uniform distribution on the circle, so the density
is a finite constant.

The cause is in `logpdf`, which guards the closed-form expression with a strict
comparison:

```python
jnp.where(lax.gt(kappa, zero), <closed-form density>, np.nan)
```

`kappa == 0` is a mathematically valid boundary value (excluded only by the
strict `>`), so it incorrectly falls into the `nan` branch. The closed-form
expression itself already evaluates correctly at `kappa == 0`, since
`bessel_i0e(0) == 1`, giving `logpdf = -log(2 * pi)`.

### Fix

Use a non-strict comparison `lax.ge(kappa, zero)` so `kappa == 0` uses the
closed-form expression. `kappa < 0` remains out of support and returns `nan`,
matching SciPy. This fixes both `pdf` (which is `exp(logpdf(...))`) and
`logpdf`.

### Verification

Reproducer from the issue, after the fix (x64 enabled):

```
       x  kappa |          scipy            jax            jit
   0.000   0.00 |     0.15915494     0.15915494     0.15915494
   1.000   0.00 |     0.15915494     0.15915494     0.15915494
  -1.000   0.00 |     0.15915494     0.15915494     0.15915494
   3.142   0.00 |     0.15915494     0.15915494     0.15915494
  -3.142   0.00 |     0.15915494     0.15915494     0.15915494
   0.000   1.00 |     0.34171049     0.34171049     0.34171049   (control: kappa>0)
     nan   0.00 |            nan            nan            nan    (control: non-finite x)
     inf   0.00 |            nan            nan            nan    (control: non-finite x)
   0.000  -1.00 |            nan            nan            nan    (control: kappa<0)
```

`logpdf(x, 0) == -log(2 * pi)` matches SciPy, and gradients w.r.t. both `x` and
`kappa` are finite at `kappa == 0` (no `nan` leaks through the `where`).

### Tests

Added `testVonMisesPdfKappaZero`, which checks `pdf`/`logpdf` at `kappa == 0`
against SciPy (the existing `testVonMises{Pdf,LogPdf}` take `abs(kappa)` and the
default RNG essentially never samples exactly `0`, so the boundary was
untested), the jitted path, and that `kappa < 0` still returns `nan`. All
existing von Mises tests continue to pass.

### CODE REVIEW

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
